carto_renderer/service.py

```diff
@@ -208,13 +210,19 @@ def post(self):
                 raise BadRequest('"zoom" must be an integer.',
                                  request_body=geobody)
 
-            path = 'http://{host}:{port}/style?style={css}'.format(
+
+            path = 'http://{host}:{port}/style'.format(
                 host=self.style_host,
-                port=self.style_port,
-                css=quote_plus(geobody['style']))
+                port=self.style_port)
 
-            tile = {layer: [base64.b64decode(wkb) for wkb in wkbs]
-                    for layer, wkbs in geobody['tile'].items()}
+            tile = {
+                layer: [
+                    {
+                        'wkbs': base64.b64decode(feature['wkbs']),
+                        'attributes': json.loads(base64.b64decode(feature['attributes']))
+                    } for feature in features
+                ] for layer, features in geobody['tile'].items()
```

Comment:
This formatting is strange, it obscures the fact that it's a for comprehension.  It looks like some sort of obscure `for` loop.